### PR TITLE
Use Python 3 instead of Python 2 (EOL'd)

### LINKED ISSUE #8093: Move away from Python2
Python2 reached EOL, we should switch to python3 in all scripts.

### PR DESCRIPTION
**Pull request by christianparpart**
This is in DRAFT still. Let's see if CI passes, and what else I may have missed.

Fixes #8093

### CODE REVIEW

File: scripts/deps-ppa/static_z3.sh [hunk 1]
```diff
@@ -76,7 +76,7 @@ Build-Depends: debhelper (>= 9.0.0),
                git,
                libgmp-dev,
                dh-python,
-               python
```
Comment (chriseth) [inline scripts/deps-ppa/static_z3.sh:79]: This script is not run by the CI, we have to try creating a ppa package locally.

File: scripts/fix_homebrew_paths_in_standalone_zip.py [hunk 2]
```diff
@@ -55,7 +54,7 @@ def readDependencies(fname):
                         command = "install_name_tool -change " + \
                             library + " @executable_path/./" + \
                             os.path.basename(library) + " " + fname
-                        print command
+                        print(command)
```
Comment (chriseth) [inline scripts/fix_homebrew_paths_in_standalone_zip.py:57]: When is this script run?

Comment (erak) [reply scripts/fix_homebrew_paths_in_standalone_zip.py:57]: It is called from `release.sh`, if run on OSX.

**Review by erak — CHANGES_REQUESTED** [review]
Looks good in general, but the code extraction from docs need some further adjustments.

File: scripts/deps-ppa/static_z3.sh (hunk 1 above)
Comment (christianparpart) [reply scripts/deps-ppa/static_z3.sh:79]: /cc @ekpyron  - I may need your advice for that :)

Comment (ekpyron) [reply scripts/deps-ppa/static_z3.sh:79]: I can check - but I'd also say it'd be fine to leave this script as is for now and only update it the next time we need to run it... (EDIT: with "as is" I mean *not* changing to python3 :-)).

**Review by chriseth — APPROVED** [review]

**Pull request merged**